app/controllers/planning_applications_controller.rb

```diff
@@ -101,14 +101,21 @@ def confirm_validation
   def validate
```

Comment:
I would really love to do a refactor which gets all this logic out of the controller and into the model or some kind of PORO (maybe a form object 😁). But when I started so many different things started to unravel that I thought I'd better leave it for another day!